Hblasinski/dev macbeth creation update

hblasins opened the pull request:
Updated the code in piCreateMacbethChart(), added various bug fixes and improvements.
@Zhenyi-Liu given that I'm changing some of the basic classes, I'd appreciate you taking a look at these changes.

hblasins committed:
Updated the piCreateMacbethChart code to support PBRT-v4.
Vaious bug fixes.

Zhenyi-Liu reviewed (changes requested):
I think line 44 in piEXR2Mat should be = instead of ==. Everything else seems good to me, thanks!

wandell commented:
Is there a validation script that we can run to see that you have achieved your goal with these changes?  That way we can always check that other changes don't interfere with what you are doing.

Also, after these changes, does v_iset3d_v4 run properly?

Finally, I should have drawn your attention to the piRecipeCreate.  We do create some scenes with Macbeth Color Checkers in them - though they may not be structured the way you want.

Thanks,
Brian

ZhengLyuFelix commented:
Hi Henryk, Sorry for a late reply - the changes look good to me. As many of the changes are related to the assets operation, making sure that the v_iset3d_v4 runs as Brian suggested would be great as it involves tests on the assets function sanity checks.

Thanks,
Zheng

hblasins committed:
Added support for underwater rendering.

hblasins committed:
Added a water medium creation function.

wandell committed:
With HB, running on GPU on orange.

Puzzled about the size of the chart in meters.
Trying different amounts of water between the camera and the chart

wandell committed:
Merge branch 'dev' into hblasinski/dev-MacbethCreationUpdate
